drivers/sensor/bmp581/bmp581.h

```diff
@@ -0,0 +1,352 @@
+#define BMP5_IIR_FLUSH_FORCED_EN_POS 2
+
+/* Effective OSR configurations and ODR valid status */
+#define BMP5_OSR_TEMP_EFF_MSK 0x07
+
+#define BMP5_OSR_PRESS_EFF_MSK 0x38
+#define BMP5_OSR_PRESS_EFF_POS 3
+
+#define BMP5_ODR_IS_VALID_MSK 0x80
+#define BMP5_ODR_IS_VALID_POS 7
+
+/* Powermode */
+#define BMP5_POWERMODE_MSK 0x03
+
+#define BMP5_DEEP_DISABLE_MSK 0x80
+#define BMP5_DEEP_DISABLE_POS 7
+
+/* Fifo configurations */
+#define BMP5_FIFO_THRESHOLD_MSK 0x1F
+
+#define BMP5_FIFO_MODE_MSK 0x20
+#define BMP5_FIFO_MODE_POS 5
+
+#define BMP5_FIFO_DEC_SEL_MSK 0x1C
+#define BMP5_FIFO_DEC_SEL_POS 2
+
+#define BMP5_FIFO_COUNT_MSK 0x3F
+
+#define BMP5_FIFO_FRAME_SEL_MSK 0x03
+
+/* Out-of-range configuration */
+#define BMP5_OOR_THR_P_LSB_MSK 0x0000FF
+
+#define BMP5_OOR_THR_P_MSB_MSK 0x00FF00
+
+#define BMP5_OOR_THR_P_XMSB_MSK 0x010000
+#define BMP5_OOR_THR_P_XMSB_POS 16
+
+/* Macro to mask xmsb value of oor threshold from register(0x35) value */
+#define BMP5_OOR_THR_P_XMSB_REG_MSK 0x01
+
+#define BMP5_OOR_COUNT_LIMIT_MSK 0xC0
+#define BMP5_OOR_COUNT_LIMIT_POS 6
+
+/* NVM configuration */
+#define BMP5_NVM_ADDR_MSK 0x3F
+
+#define BMP5_NVM_PROG_EN_MSK 0x40
+#define BMP5_NVM_PROG_EN_POS 6
+
+#define BMP5_NVM_DATA_LSB_MSK 0x00FF
+
+#define BMP5_NVM_DATA_MSB_MSK 0xFF00
+
+/*!
+ * @brief OSR, ODR and pressure configuration structure
+ */
+struct bmp581_osr_odr_press_config {
+	/*! Temperature oversampling
+	 * Assignable macros :
+	 * - BMP5_OVERSAMPLING_1X
+	 * - BMP5_OVERSAMPLING_2X
+	 * - BMP5_OVERSAMPLING_4X
+	 * - BMP5_OVERSAMPLING_8X
+	 * - BMP5_OVERSAMPLING_16X
+	 * - BMP5_OVERSAMPLING_32X
+	 * - BMP5_OVERSAMPLING_64X
+	 * - BMP5_OVERSAMPLING_128X
+	 */
+	uint8_t osr_t;
+
+	/*! Pressure oversampling
+	 * Assignable macros :
+	 * - BMP5_OVERSAMPLING_1X
+	 * - BMP5_OVERSAMPLING_2X
+	 * - BMP5_OVERSAMPLING_4X
+	 * - BMP5_OVERSAMPLING_8X
+	 * - BMP5_OVERSAMPLING_16X
+	 * - BMP5_OVERSAMPLING_32X
+	 * - BMP5_OVERSAMPLING_64X
+	 * - BMP5_OVERSAMPLING_128X
+	 */
+	uint8_t osr_p;
+
+	/*! Enable pressure
+	 * BMP5_ENABLE  = Enables pressure data
+	 * BMP5_DISABLE = Disables pressure data
+	 */
+	uint8_t press_en;
+
+	/*! Output Data Rate */
+	uint8_t odr;
+};
+
+struct bmp581_sample {
+	float pressure;
+	float temperature;
+};
+
+struct bmp581_data {
+	const struct device *i2c;
+#ifdef CONFIG_BMP581_TRIGGER
+	const struct device *dev;
+	struct gpio_dt_spec gpio;
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+	sensor_trigger_handler_t drdy_handler;
+#endif
+	uint8_t i2c_addr;
+	uint8_t chip_id;
+	struct bmp581_sample last_sample;
+	struct bmp581_osr_odr_press_config osr_odr_press_config;
+};
+
+struct bmp581_config {
+	struct i2c_dt_spec i2c;
+	uint16_t i2c_addr;
+#ifdef CONFIG_BMP581_TRIGGER
+	struct gpio_dt_spec input;
+#endif
+};
+
+#ifdef CONFIG_BMP581_TRIGGER
+int bmp581_trigger_init(const struct device *dev);
+int bmp581_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+		       sensor_trigger_handler_t handler);
```

Comment:
Trigger support doesn't appear to be implemented yet. Please remove all trigger references until this feature is actually imoplemented

Reply:
Indeed, I have just removed the not implemented parts from the source.